codegen/src/TypeMapper.scala

```diff
@@ -1,35 +1,56 @@
 package besom.codegen
 
+import besom.codegen.Config.ProviderConfig
+
 import scala.util.matching.Regex
 import scala.meta._
 import scala.meta.dialects.Scala33
 import besom.codegen.metaschema._
 
+case class TypeToken private (provider: String, module: String, `type`: String, raw: String) {
```

Comment:
The design of this class looks slightly suspicious and against the usual conventions about case classes.
I would recommend not to override `toString` - if we use it just for debugging it would be better not to lose the information about the original string. Otherwise one might get surprised that `foo:bar:Baz` and `foo:bar:Baz` are not equal (because the `raw` part was different). If we need to make a string from that as a part of business logic - let's make this explicit by choosing a different method name for clarity.
Also I'm not sure if it's a good idea to use the extractor if all the fields are strings and we rely just on the order of fields - something's likely to go wrong if some code around this class gets refactored - it would be better to rely on named field access

Reply:
Thank you I will remove the `toString` and make the `apply` always create a correct object